"The new element doesn't match the prefix"

I'm quite frequently getting the following error:

```
java.lang.AssertionError: The new element doesn't match the prefix
	at com.intellij.codeInsight.completion.CompletionResult.withLookupElement(CompletionResult.java:59)
	at com.intellij.codeInsight.completion.XmlCompletionContributor$3.lambda$addCompletions$0(XmlCompletionContributor.java:131)
	at com.intellij.codeInsight.completion.CompletionResultSet$1.consume(CompletionResultSet.java:161)
	at com.intellij.codeInsight.completion.CompletionResultSet$1.consume(CompletionResultSet.java:148)
	at com.intellij.codeInsight.completion.CompletionResultSet.passResult(CompletionResultSet.java:56)
	at com.intellij.codeInsight.completion.impl.CompletionServiceImpl$CompletionResultSetImpl.addElement(CompletionServiceImpl.java:148)
	at com.intellij.codeInsight.completion.CompletionResultSet.addAllElements(CompletionResultSet.java:86)
	at com.intellij.codeInsight.completion.impl.CompletionServiceImpl$CompletionResultSetImpl.lambda$addAllElements$0(CompletionServiceImpl.java:134)
	at com.intellij.codeInsight.completion.CompletionThreadingBase.withBatchUpdate(CompletionThreadingBase.java:33)
	at com.intellij.codeInsight.completion.impl.CompletionServiceImpl$CompletionResultSetImpl.addAllElements(CompletionServiceImpl.java:134)
	at TabNineCompletionContributor.fillCompletionVariants(TabNineCompletionContributor.java:219)
	at com.intellij.codeInsight.completion.CompletionService.getVariantsFromContributors(CompletionService.java:63)
	at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:148)
	at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:141)
	at com.intellij.codeInsight.completion.XmlCompletionContributor$3.addCompletions(XmlCompletionContributor.java:126)
	at com.intellij.codeInsight.completion.CompletionProvider.addCompletionVariants(CompletionProvider.java:40)
	at com.intellij.codeInsight.completion.CompletionContributor.fillCompletionVariants(CompletionContributor.java:150)
	at com.intellij.codeInsight.completion.XmlCompletionContributor.fillCompletionVariants(XmlCompletionContributor.java:181)
	at com.intellij.codeInsight.completion.CompletionService.getVariantsFromContributors(CompletionService.java:63)
	at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:148)
	at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:141)
	at com.intellij.codeInsight.template.impl.LiveTemplateCompletionContributor$1.addCompletions(LiveTemplateCompletionContributor.java:77)
	at com.intellij.codeInsight.completion.CompletionProvider.addCompletionVariants(CompletionProvider.java:40)
	at com.intellij.codeInsight.completion.CompletionContributor.fillCompletionVariants(CompletionContributor.java:150)
	at com.intellij.codeInsight.completion.CompletionService.getVariantsFromContributors(CompletionService.java:63)
	at com.intellij.codeInsight.completion.CompletionService.performCompletion(CompletionService.java:119)
	at com.intellij.codeInsight.completion.impl.CompletionServiceImpl.performCompletion(CompletionServiceImpl.java:55)
	at com.intellij.codeInsight.completion.CompletionProgressIndicator.calculateItems(CompletionProgressIndicator.java:824)
	at com.intellij.codeInsight.completion.CompletionProgressIndicator.runContributors(CompletionProgressIndicator.java:809)
	at com.intellij.codeInsight.completion.CodeCompletionHandlerBase.lambda$null$5(CodeCompletionHandlerBase.java:325)
	at com.intellij.codeInsight.completion.AsyncCompletion.lambda$tryReadOrCancel$5(CompletionThreading.java:172)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1106)
	at com.intellij.codeInsight.completion.AsyncCompletion.tryReadOrCancel(CompletionThreading.java:170)
	at com.intellij.codeInsight.completion.CodeCompletionHandlerBase.lambda$startContributorThread$6(CodeCompletionHandlerBase.java:317)
	at com.intellij.codeInsight.completion.AsyncCompletion.lambda$null$0(CompletionThreading.java:95)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:169)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:591)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:537)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:59)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:156)
	at com.intellij.codeInsight.completion.AsyncCompletion.lambda$startThread$1(CompletionThreading.java:91)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.run(ApplicationImpl.java:294)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

This is just from regular use, no idea what exactly causes it.

---

PhpStorm 2019.2.3
Build #PS-192.6817.20, built on September 25, 2019
Runtime version: 11.0.4+10-b304.69 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.12.6
GC: G1 Young Generation, G1 Old Generation
Memory: 1500M
Cores: 8
Registry: run.processes.with.pty=TRUE, debugger.watches.in.variables=false
Non-Bundled Plugins: ColorIde, com.tabnine.TabNine, NEON support, org.sylfra.idea.plugins.linessorter, ru.adelf.idea.dotenv

TabNine: v0.1.3

Plugin version 0.1.4 in following IDE:
```
IntelliJ IDEA 2019.3.1 (Ultimate Edition)
Build #IU-193.5662.53, built on December 18, 2019
Subscription is active until January 9, 2021
Runtime version: 11.0.5+10-b520.17 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.15.2
GC: ParNew, ConcurrentMarkSweep
Non-Bundled Plugins: com.intellij.plugins.watcher, org.jetbrains.plugins.rest, Pythonid, awesome.console, com.dubreuia, com.github.mfedko.idea.plugins.filelanguage, org.jetbrains.plugins.go-template, com.intellij.kubernetes, com.intellij.lang.puppet, com.jetbrains.plugins.ini4idea, com.tabnine.TabNine, ideanginx9, mobi.hsz.idea.gitignore, org.intellij.plugins.hcl, org.jetbrains.plugins.go, org.jetbrains.plugins.ruby, org.sylfra.idea.plugins.linessorter, org.toml.lang
```
Error:
```
java.lang.AssertionError: The new element doesn't match the prefix
	at com.intellij.codeInsight.completion.CompletionResult.withLookupElement(CompletionResult.java:45)
	at com.intellij.codeInsight.completion.CompletionService$1.consume(CompletionService.java:111)
	at com.intellij.codeInsight.completion.CompletionService$1.consume(CompletionService.java:93)
	at com.intellij.codeInsight.completion.CompletionResultSet.passResult(CompletionResultSet.java:56)
	at com.intellij.codeInsight.template.impl.LiveTemplateCompletionContributor$1.lambda$addCompletions$0(LiveTemplateCompletionContributor.java:78)
	at com.intellij.codeInsight.completion.CompletionResultSet$1.consume(CompletionResultSet.java:161)
	at com.intellij.codeInsight.completion.CompletionResultSet$1.consume(CompletionResultSet.java:148)
	at com.intellij.codeInsight.completion.CompletionResultSet.passResult(CompletionResultSet.java:56)
	at com.intellij.codeInsight.completion.impl.CompletionServiceImpl$CompletionResultSetImpl.addElement(CompletionServiceImpl.java:148)
	at com.intellij.codeInsight.completion.CompletionResultSet.addAllElements(CompletionResultSet.java:86)
	at com.intellij.codeInsight.completion.impl.CompletionServiceImpl$CompletionResultSetImpl.lambda$addAllElements$0(CompletionServiceImpl.java:134)
	at com.intellij.codeInsight.completion.CompletionThreadingBase.withBatchUpdate(CompletionThreadingBase.java:33)
	at com.intellij.codeInsight.completion.impl.CompletionServiceImpl$CompletionResultSetImpl.addAllElements(CompletionServiceImpl.java:134)
	at TabNineCompletionContributor.fillCompletionVariants(TabNineCompletionContributor.java:222)
	at com.intellij.codeInsight.completion.CompletionService.getVariantsFromContributors(CompletionService.java:63)
	at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:148)
	at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:141)
	at com.intellij.codeInsight.template.impl.LiveTemplateCompletionContributor$1.addCompletions(LiveTemplateCompletionContributor.java:77)
	at com.intellij.codeInsight.completion.CompletionProvider.addCompletionVariants(CompletionProvider.java:26)
	at com.intellij.codeInsight.completion.CompletionContributor.fillCompletionVariants(CompletionContributor.java:150)
	at com.intellij.codeInsight.completion.CompletionService.getVariantsFromContributors(CompletionService.java:63)
	at com.intellij.codeInsight.completion.CompletionService.performCompletion(CompletionService.java:122)
	at com.intellij.codeInsight.completion.impl.CompletionServiceImpl.performCompletion(CompletionServiceImpl.java:55)
	at com.intellij.codeInsight.completion.CompletionProgressIndicator.calculateItems(CompletionProgressIndicator.java:827)
	at com.intellij.codeInsight.completion.CompletionProgressIndicator.runContributors(CompletionProgressIndicator.java:812)
	at com.intellij.codeInsight.completion.CodeCompletionHandlerBase.lambda$null$5(CodeCompletionHandlerBase.java:326)
	at com.intellij.codeInsight.completion.AsyncCompletion.lambda$tryReadOrCancel$5(CompletionThreading.java:172)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:997)
	at com.intellij.codeInsight.completion.AsyncCompletion.tryReadOrCancel(CompletionThreading.java:170)
	at com.intellij.codeInsight.completion.CodeCompletionHandlerBase.lambda$startContributorThread$6(CodeCompletionHandlerBase.java:318)
	at com.intellij.codeInsight.completion.AsyncCompletion.lambda$null$0(CompletionThreading.java:95)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:163)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:585)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:531)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:59)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:150)
	at com.intellij.codeInsight.completion.AsyncCompletion.lambda$startThread$1(CompletionThreading.java:91)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.run(ApplicationImpl.java:238)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

--- a/src/main/java/com/tabnine/TabNineCompletionContributor.java
+++ b/src/main/java/com/tabnine/TabNineCompletionContributor.java
@@ -113,6 +113,9 @@ public class TabNineCompletionContributor extends CompletionContributor {
         public boolean prefixMatches(@NotNull LookupElement element) {
             if (element instanceof TabNineLookupElement) {
                 return true;
+            } else if (element instanceof LookupElementDecorator) {
+                LookupElementDecorator decorator = (LookupElementDecorator) element;
+                return prefixMatches(decorator.getDelegate());
             }
             return super.prefixMatches(element);
         }

